packages/launchpad/src/signer.ts

```diff
@@ -34,5 +61,5 @@ export interface OfflineSigner {
    * @param signerAddress The address of the account that should sign the transaction
    * @param signDoc The content that should be signed
    */
-  readonly sign: (signerAddress: string, signDoc: StdSignDoc) => Promise<SignResponse>;
+  readonly sign: (signerAddress: string, signDoc: T["signJob"]) => Promise<T["signResponse"]>;
```

Comment:
what happens to this signature if we implement `OfflineSigner<AminoSignSpec | DirectSignSpec>`?

Would it be:

`readonly sign: (signerAddress: string, signDoc: AminoSignJob | DirectSignJob) => Promise<AminoSignResponse | DirectSignResponse>;`
	
Or a compiler error?

Reply:
Yes that's fine. In fact I was hoping it would match up the inputs and outputs but this compiles:
```ts
  public async sign(
    _signerAddress: string,
    _signJob: AminoSignJob | DirectSignJob,
  ): Promise<DirectSignResponse> {
    throw new Error("xxx");
  }
```